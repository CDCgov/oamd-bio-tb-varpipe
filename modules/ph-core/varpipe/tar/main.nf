process VARPIPE_TAR {

    label 'process_single'

    container 'oamd-bio-nf-core-ubuntu:24.04_4da1a92_v3'

    input:
    val(metas)
    path(files)
    val(date)

    output:
    path("*.tar.gz"), emit: varpipe_zip
    path("versions.yml"), emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "varpipe_outputs_${date}"
    def stage_files = [
        metas.collect { "mkdir -p ${prefix}/${it.id}" },
        metas.collect { "mv ${it.filenames.join(' ')} ${prefix}/${it.id}" }
    ].flatten().join(" && ")
    """
    ${stage_files}

    tar \\
        -czh \\
        -f ${prefix}.tar.gz \\
        ${prefix} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        tar: \$(echo \$(tar --version 2>&1) | sed 's/^.*(GNU tar) //; s/ Copyright.*\$//')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "varpipe_outputs_${date}"
    """
    touch ${prefix}.tar.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        tar: \$(echo \$(tar --version 2>&1) | sed 's/^.*(GNU tar) //; s/ Copyright.*\$//')
    END_VERSIONS
    """
}
