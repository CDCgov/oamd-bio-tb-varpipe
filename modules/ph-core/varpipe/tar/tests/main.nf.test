nextflow_process {

    name "Test Process VARPIPE_TAR"
    script "../main.nf"
    process "VARPIPE_TAR"

    tag "modules"
    tag "modules_phcore"
    tag "varpipe"
    tag "varpipe/tar"

    test("tar [varpipe outputs]") {

        when {
            process {
                """
                // Prepare input data
                def metas = [
                    [ id:'sample1', filenames: [file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/sample1_mapped.txt'), file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/sample1_unmapped.txt')] ],
                    [ id:'sample2', filenames: [file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/sample2_mapped.txt'), file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/sample2_unmapped.txt')] ]
                ]

                input[0] = metas
                input[1] = metas.collectMany { meta -> meta.filenames }
                input[2] = 'TEST_DATE'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.varpipe_zip.size() > 0 },
                { assert snapshot(process.out.versions).match() }
            )
        }
    }

    test("tar [varpipe outputs] -stub") {

        options "-stub"

        when {
            process {
                """
                // Prepare input data
                def metas = [
                    [ id:'sample1', filenames: [file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/sample1_mapped.txt'), file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/sample1_unmapped.txt')] ],
                    [ id:'sample2', filenames: [file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/sample2_mapped.txt'), file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/sample2_unmapped.txt')] ]
                ]

                input[0] = metas
                input[1] = metas.collectMany { meta -> meta.filenames }
                input[2] = 'TEST_DATE'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.varpipe_zip.size() > 0 },
                { assert snapshot(process.out.versions).match() }
            )
        }
    }
}

