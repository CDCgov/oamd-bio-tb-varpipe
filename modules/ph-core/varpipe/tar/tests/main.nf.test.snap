{
    "tar [varpipe outputs]": {
        "content": [
            [
                "versions.yml:md5,6bcc102211483ec6c5ce5e268e277e58"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-23T16:13:45.656564067"
    },
    "tar [varpipe outputs] -stub": {
        "content": [
            [
                "versions.yml:md5,6bcc102211483ec6c5ce5e268e277e58"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-23T16:13:51.024401421"
    }
}
